lib/topologies/mongos.js

```diff
@@ -661,17 +662,11 @@ function applyAuthenticationContexts(self, server, callback) {
   // Apply one of the contexts
   function applyAuth(authContexts, server, callback) {
     if (authContexts.length === 0) return callback();
-    // Get the first auth context
-    var authContext = authContexts.shift();
-    // Copy the params
-    var customAuthContext = authContext.slice(0);
-    // Push our callback handler
-    customAuthContext.push(function(/* err */) {
-      applyAuth(authContexts, server, callback);
-    });
+    // Get credentials
+    const credentials = authContexts.shift();
```

Comment:
should `authContexts` be renamed? This looks weird that we shift a credential off the auth context